django/db/migrations/autodetector.py

```diff
@@ -722,8 +722,8 @@ def generate_deleted_models(self):
         for app_label, model_name in all_deleted_models:
             model_state = self.from_state.models[app_label, model_name]
             model = self.old_apps.get_model(app_label, model_name)
-            if not model._meta.managed:
-                # Skip here, no need to handle fields for unmanaged models
+            if not model._meta.managed and not deleted_unmanaged_models:
+                # Skip here, no need to handle only fields for unmanaged models
                 continue
```

Comment:
Do the tests pass if you completely remove this `if` statement? That looks like it would be more appropriate in this case as I can see how it can ever be `True` at this point.

Reply:
yes they pass, i agree the if statement can be removed